gitops-application.go

```diff
@@ -38,15 +38,24 @@ func applyGitopsApplications(c *cli.Context) error {
 		"clusterIdentifier": clusterIdentifier,
 		"repoIdentifier":    repoIdentifier,
 	})
+
+	applicationName := valueToString(GetNestedValue(requestBody, "gitops", "name").(string))
+	syncApplicationURL := GetUrlWithQueryParams("", baseURL,
+		fmt.Sprintf(GITOPS_APPLICATION_ENDPOINT+"/%s", applicationName+"/sync"), map[string]string{
```

Comment:
move entire string to constant and pass variable:
```
const GITOPS_APPLICATION_ENDPOINT = "applications/%s/sync"
 fmt.Sprintf(GITOPS_APPLICATION_ENDPOINT, applicationName)
 ```

Reply:
`/sync` endpoint comes in only for the Sync operation and `GITOPS_APPLICATION_ENDPOINT` is also used in three other places so it's safe to skip from the constant.

Reply:
Let's declare the `sync` in the same way as `GITOPS_APPLICATION_ENDPOINT` and use it here like:
`fmt.Sprintf(GITOPS_APPLICATION_ENDPOINT+"/%s", applicationName+GITOPS_SYNC_ENDPOINT)`

Adding it in the constants also gives reader an opportunity to see that this endpoint is being used somewhere in GitOps flow which goes missing if we just hard-code it.

Reply:
IMO we use CONSTANTS as an easy way around VARIABLES (i.e. for things that can be changed/modified as per user needs) and adding more constants that too for `/sync` only increases the list.

Maybe we can use code comments to help the contributors of the future.